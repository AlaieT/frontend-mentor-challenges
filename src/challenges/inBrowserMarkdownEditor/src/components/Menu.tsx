import React, { useMemo } from "react";

import { ReactComponent as IconDocument } from "../assets/images/icon-document.svg";
import { ReactComponent as IconDarkMode } from "../assets/images/icon-dark-mode.svg";
import { ReactComponent as IconLightMode } from "../assets/images/icon-light-mode.svg";
import styles from "../styles/components/menu.module.scss";

import type { MenuProps } from "../types";

const Menu = ({ myDocuments, mode, handleMode }: MenuProps) => {
  const myDocumentsList = useMemo(
    () =>
      Object.keys(myDocuments).map((key) => ({
        name: key,
        date: myDocuments[key].date
      })),
    [myDocuments]
  );

  return (
    <nav id={styles.menu}>
      <div id={styles["menu-body"]}>
        <div id={styles["menu-title"]}>MY DOCUMENTS</div>
        <button id={styles["new-document"]}>+ New Document</button>
        <ul id={styles.documents}>
          {myDocumentsList.map(({ name, date }) => (
            <li key={name}>
              <IconDocument />
              <p>{date}</p>
              <p>{name}.md</p>
            </li>
          ))}
        </ul>
      </div>
      <div id={styles["mode-controls"]}>
        <IconDarkMode />
        <label id={styles["toggle-switch"]}>
          <input
            type="checkbox"
            defaultChecked
            onChange={() => handleMode(mode === "light" ? "dark" : "light")}
          />
          <span />
        </label>
        <IconLightMode />
      </div>
    </nav>
  );
};

export default Menu;
