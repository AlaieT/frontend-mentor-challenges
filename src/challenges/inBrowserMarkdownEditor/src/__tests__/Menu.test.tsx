import React from "react";
import { render, cleanup } from "@testing-library/react";

import Menu from "../components/Menu";

afterEach(cleanup);

describe("Menu", () => {
  describe("correct render", () => {
    it("should match snapshot", () => {
      expect(
        render(
          <Menu
            myDocuments={{
              "untitled-document": { date: "01 April 2022", text: "" },
              welcome: { date: "01 April 2022", text: "" }
            }}
            mode="dark"
          />
        ).asFragment()
      ).toMatchSnapshot();
    });
  });
});
