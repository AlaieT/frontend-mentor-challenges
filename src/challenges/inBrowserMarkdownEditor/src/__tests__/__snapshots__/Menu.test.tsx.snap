// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu correct render should match snapshot 1`] = `
<DocumentFragment>
  <nav
    id="menu"
  >
    <div
      id="menu-body"
    >
      <div
        id="menu-title"
      >
        MY DOCUMENTS
      </div>
      <button
        id="new-document"
      >
        + New Document
      </button>
      <ul
        id="documents"
      >
        <li>
          <div />
          <p>
            01 April 2022
          </p>
          <p>
            untitled-document.md
          </p>
        </li>
        <li>
          <div />
          <p>
            01 April 2022
          </p>
          <p>
            welcome.md
          </p>
        </li>
      </ul>
    </div>
    <div
      id="mode-controls"
    >
      <div />
      <label
        id="toggle-switch"
      >
        <input
          checked=""
          type="checkbox"
        />
        <span />
      </label>
      <div />
    </div>
  </nav>
</DocumentFragment>
`;
